storage/rocksdb/rdb_datadic.cc

```diff
@@ -3137,6 +3147,38 @@ bool Rdb_tbl_def::put_dict(Rdb_dict_manager *const dict,
   return false;
 }
 
+// Length that each index flag takes inside the record.
+// Each index in the array maps to the enum INDEX_FLAG
+static const std::array<int, 1> index_flag_lengths = {
+    {ROCKSDB_SIZEOF_TTL_RECORD}};
+
+
+bool Rdb_key_def::has_index_flag(uint32 index_flags, enum INDEX_FLAG flag) {
+  return flag & index_flags;
+}
+
+uint32 Rdb_key_def::calculate_index_flag_offset(uint32 index_flags,
+                                                enum INDEX_FLAG flag) {
+
+  DBUG_ASSERT(Rdb_key_def::has_index_flag(index_flags, flag));
+
+  uint offset = 0;
+  for (size_t bit = 0; bit < sizeof(index_flags) * CHAR_BIT; ++bit) {
+    int mask = 1 << bit;
+
+    /* Exit once we've reached the proper flag */
+    if (flag == mask) {
```

Comment:
Hmm...what do you mean by multiple flags? This is the exit condition where the `flag == mask` the mask should always only have 1 bit set, and the enum `flag` passed is an enum where only 1 bit can be turned on.  I guess it doesn't hurt to turn it into `&`, but that makes it seem like multiple bits can be turned on when only 1 can.